src/auth/auth.controller.ts

```diff
@@ -51,42 +60,52 @@ export class AuthController {
   }
 
   @Get('refresh')
-  async refresh(@Req() request, @Res({ passthrough: true }) response){
+  async refresh(@Req() request, @Res({ passthrough: true }) response, @Headers('User-Agent') headers){
     const tokenFromUser = request.cookies.refreshToken;
+    const parsedUA = parser(headers);
 
-    const { refreshToken, ...rest } = await this.authService.refreshSession(tokenFromUser);
+    const { refreshToken, ...rest } = await this.authService.refreshSession(tokenFromUser, parsedUA);
 
     response.cookie( 'refreshToken', refreshToken );
     return rest;
   }
 
+  @Get('sessions')
+  @UseGuards(TokenAuthGuard)
+  getSession(@Req() request){
+    return this.sessionHandler.getAllSession(request.user.id);
+  }
+
   @Get('oauth/google')
   @UseGuards(AuthGuard('google'))
-  async googleAuth(@Req() req){}
+  googleAuth(){}
 
   @Get('oauth/callback/google')
   @UseGuards(AuthGuard('google'))
-  async googleAuthRedirect(@Req() request, @Res({ passthrough: true }) response): Promise<object>{
-
-    const { refreshToken, ...rest } = await this.sessionHandler.getSession(request.user.id);
+  async googleAuthRedirect(@Req() request, @Res({ passthrough: true }) response, @Headers('User-Agent') headers): Promise<object>{
 
-    response.cookie( 'refreshToken', refreshToken );
+    const parsedUA = parser(headers);
```

Comment:
make decorator for this parser, don't copy paste the same code for each function)


`googleAuthRedirect(@Req() request, @SystemInfo() systemInfo) Promise<Session>{}`